composer/datasets/in_context_learning_evaluation.py

```diff
@@ -103,117 +108,171 @@ def _get_fewshot_sample_idxs(dataset_size: int, num_fewshot: int, sample_idx: in
-        tokenizer (Union[transformers.PreTrainedTokenizer, transformers.PreTrainedTokenizerFast]): The tokenizer used to map between strings and token ids
-        batch_size (int): Size of a batch used for eval
-        max_seq_len (int): The maximum sequence length supported by the model
-        pad_tok_id (int): The special token reserved for padding batches
-        num_fewshot (int): The number of complete fewshot examples to prepend before each test example
-        prompt_string (str): Prompt string to put once before all fewshot examples/test examples (e.g. 'translate english to french')
-        example_delimiter (str): Separator that goes between individual (context, answer) pairs (e.g. '\n')
-        continuation_delimiter: (str): Separator that goes between context and answer in each example (e.g. '\nA: ')
-        destination_path (str): Temporary path to store downloaded datasets
-        question_prelimiter (str): String to put before each question (e.g. 'Q: ')
-        fewshot_random_seed (int): Random seed to use for fewshot sampling
-    """
+        self.prelimiter = prelimiter
+        self.example_delimiter = example_delimiter
+        self.continuation_delimiter = continuation_delimiter
+        self.context_key = context_key
+        self.answer_key = answer_key
+        self.stacked_keys = stacked_keys
+
+        if hf_parsing_func is not None:
+            self._parse_hf_dataset = hf_parsing_func
+        else:
+            self._parse_hf_dataset = lambda example, **kwargs: {
+                k: ' '.join([str(example[col]) for col in v]) for k, v in hf_parsing_vars.items()
+            }
+        self.dataset = self._read_dataset(dataset_uri, destination_path, hf_loading_vars, hf_parsing_vars)
+        self.strip_data = strip_dataset
+        if self.strip_data:
+            self.dataset = self.dataset.map(strip_data)
 
-    def _read_dataset(self, dataset: Dataset) -> List[Dict[str, str]]:
-        result = []
-        for example in dataset:
-            result.append({
-                'context': example['context'],
-                'answer': example['answer'],
-                'aliases': set([example['answer']] + example.get('aliases', [])),
-                'chain_of_thought': example.get('chain_of_thought', '')
-            })
-        return result
+        fewshot_rng = random.Random(fewshot_random_seed)
+        self.encoded_dataset = self.dataset.map(
+            self._prep_example,
+            with_indices=True,
+            fn_kwargs={
+                'num_fewshot': num_fewshot,
+                'prompt_string': prompt_string,
+                'fewshot_rng': fewshot_rng,
+            },
+        )
 
-    def __init__(
+    def __getitem__(self, index: int):
+        return self.encoded_dataset[index]
+
+    def __len__(self):
+        return len(self.encoded_dataset)
+
+    def get_num_samples_in_batch(self, batch: dict) -> int:
```

Comment:
[Daniel] check whether this was the same in all datasets

Reply:
https://github.com/mosaicml/composer/blob/dev/composer/datasets/in_context_learning_evaluation.py#L276
https://github.com/mosaicml/composer/blob/dev/composer/datasets/in_context_learning_evaluation.py#L460
https://github.com/mosaicml/composer/blob/dev/composer/datasets/in_context_learning_evaluation.py#L635
https://github.com/mosaicml/composer/blob/dev/composer/datasets/in_context_learning_evaluation.py#L1057